src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -558,7 +559,7 @@ private static class FieldFrame {
         private final Set<FieldFrame> children = new HashSet<>();
 
         /** Set of fields. */
-        private final Set<DetailAST> fields = new HashSet<>();
+        private final Set<DetailAST> fields = new LinkedHashSet<>();
 
```

Comment:
It is hard to justify this update to just let pitest be happy.
:(
Any other ideas?

If not I more in favor to keep it in unstable list of survivals.

Reply:
@pbludov , @strkkk , @rnveach , please share your ideas

Reply:
If we do not use any hash benefits, can it be List ?

Reply:
We can use a HashMap, with field name as key and DetailAST as value. This would be performance improvement, which gives us another reason for update.

Reply:
> It is hard to justify this update to just let pitest be happy.

I can understand that, it was working but there wasn't any proof to completely justify it.

> We can use a HashMap, with field name as key and DetailAST as value. This would be performance improvement, which gives us another reason for update.

Good idea, implemented.